client/js/components/truncate-text.js

```diff
@@ -0,0 +1,44 @@
+import fromEvent from 'xstream/extra/fromEvent';
+
+const truncateText = (caption) => {
+  const truncateClass = 'captioned-image__caption-text--truncate';
+  const moreText = '+ More';
+  const lessText = '- Less';
+  const toggleTruncate = function(isTruncated, control, className) {
```

Comment:
Could this be pulled out as a function so as not to be created everytime and save memory?